### PR TITLE
feat(input): add textinput

### PR DESCRIPTION
**Pull request by Xzirez**
- Adds text input component

Note: there were dependency issues that delayed this component. It's not hard to make :D

### CODE REVIEW

File: src/components/inputs/InputText/index.tsx [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
-function InputText() {
-  return <styled.Layout>WIP</styled.Layout>;
+import ErrorMessage from "components/display/ErrorMessage";
+
+function InputText(
+  {
+    error = null,
+    disabled = false,
+    placeholder = "Please enter a value",
+    label = "Withdraw address",
```
Comment (cwagner22) [inline src/components/inputs/InputText/index.tsx:15]: We need to add onChange, onBlur, name, ref to work with react-hook-form. See https://github.com/zignaly-open/zignaly-ui/pull/35

Comment (Xzirez) [reply src/components/inputs/InputText/index.tsx:15]: Ok, will do

**Comment by cwagner22** [comment]
Thanks, it'd be great if we could support right child components like this:
image

**Comment by Xzirez** [comment]
@cwagner22 Okay i can add this too.

**Comment by Xzirez** [comment]
Do you need the type parameter to be passed too like in #35 ? @cwagner22

**Comment by cwagner22** [comment]
@Xzirez Yes the type too so I can pass for example type=password

**Comment by Xzirez** [comment]
@cwagner22 check now if it's okay with all code added. Im unsure of the type:

type?: React.ButtonHTMLAttributes<HTMLInputElement>["type"];

Note: here tests would help loads. @tastafur will make a POC and then I will make sure to test everything after.

**Comment by NataliaAvila-PM** [comment]
Hi Vetle.

I believe we can merge this one as a starting point for the input text but for the read/copy version we need to review some details. 

The version with the icon to copy on the side should have the text already written and the user will read, and copy the information instead of typing it. Right now to me, the copy button is not making any effect.



Additionally, I believe we can add as improvements for this component the option to add the label. I will add that as a separate task.

Let me know any comments or questions.

**Comment by cwagner22** [comment]
I have already merged but please take a look at the pending comments I made. Also rightSideElement should be a component to have control over when the user presses it. So I can call it like:
```
<InputText rightSideElement={<IconButton icon={...} onClick={handleClick} />} />
```

And I think we need to make it controlled like what @german-schneck did for the InputAmount [here](https://github.com/zignaly-open/zignaly-ui/pull/40). So I can set the `value` programatically.
In the deposit modal, I need to fill it with the address for example. In that case we'll also need to support `readOnly: true`

**Comment by Xzirez** [comment]
"And I think we need to make it controlled like what @german-schneck did for the InputAmount https://github.com/zignaly-open/zignaly-ui/pull/40. So I can set the value programmatically."

@cwagner22 Im not sure what you mean by this. Do you want me to create a smaller component you can pass to the rightSideElement that simply copies the value of the input field for the user or is this a component you will create outside of zignaly-ui and pass to the rightSideElement?

**Pull request merged**